cookiecutter.json

```diff
@@ -11,7 +11,7 @@
     "false"
   ],
   "project_version": "1.0.0",
-  "dbt_version": "1.0.4",
+  "dbt_version": "1.0.5",
```

Comment:
This is cool! (just catching up on https://github.com/dbt-labs/dbt-database-adapter-scaffold/pull/25 via this auto-generated PR)

As long as this line uses the compability operator `~=`, we don't need this GHA for every _patch_ release:
https://github.com/dbt-labs/dbt-database-adapter-scaffold/blob/ad3a9855631d50f301214d797e9f379c956514fc/%7B%7Bcookiecutter.project_name%7D%7D/setup.py#L20

However, when there's a new _minor_ release of `dbt-core`, we'll need to bump both `project_version` + `dbt_version`.

A few notes:
- **MUST:** Because `project_version` is used as [the package version in `setup.py`](https://github.com/dbt-labs/dbt-database-adapter-scaffold/blob/ad3a9855631d50f301214d797e9f379c956514fc/%7B%7Bcookiecutter.project_name%7D%7D/setup.py#L6), and `dbt_version` is used in [`__version__.py`](https://github.com/dbt-labs/dbt-database-adapter-scaffold/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/dbt/adapters/%7B%7Bcookiecutter.directory_name%7D%7D/__version__.py#L1), `pip install dbt-<thisplugin>` and `dbt --version` are going to show different output. I think that's confusing, and we should use `project_version` for both.
- **COULD:** As I mentioned in https://github.com/dbt-labs/dbt-database-adapter-scaffold/pull/24#issuecomment-1099969958, I think it would be simpler to combine `project_version` + `dbt_version` into one single version. That single version should be `1.X.0`, where X is latest minor available. Users can always modify as they see fit, if they want to differ from our recommendations. To @ChenyuLInx's point, we'll want to call out in the README that a plugin will be generated at major version v1, and yet still be brand-new.